core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/downscale/StackDownscaleService.java

```diff
@@ -92,15 +86,14 @@ public void finishStackDownscale(StackScalingFlowContext context, String instanc
         flowMessageService.fireEventAndLog(stack.getId(), AVAILABLE.name(), STACK_DOWNSCALE_SUCCESS, String.join(",", deletedInstanceIds));
     }
 
-    private void cleanupDnsRecords(Stack stack, List<String> fqdns) {
+    private void cleanupDnsRecords(Stack stack, Collection<InstanceMetaData> instanceMetaDatas) {
+        Set<String> fqdns = instanceMetaDatas.stream().map(InstanceMetaData::getDiscoveryFQDN).collect(Collectors.toSet());
```

Comment:
I am wondering if it might be possible for `getDiscoveryFQDN` to return null some error cases. I think this might be called for cleanup of a cluster that might have a node that didn't successfully complete the registration. A `filter(Objects:nonNull)` might be helpful here.

I know later in the case there is `Objects.requireNonNull(hostNames, "Hostnames must be set");` so it will throw an exception. But I wonder if this validation is called for the cluster deletion if it will prevent a cleanup from completing without the "force".

Reply:
we are ignoring cleanup error, but sure, I can add the filter